### PR TITLE
Add initial rspec test framework

### PR DESCRIPTION
**Pull request by digitalronin**
* Extract a FileFetcher class (for easier testing)
* Fix the license, which is used by the tests
* Extract Checker superclass for checks
* Extract RequiredFileChecker

### CODE REVIEW

File: spec/license_checker_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
+require_relative 'spec_helper'
+
+describe Repoman::LicenseChecker do
+  let(:repo) { double(Github::Mash, { full_name: 'dummy-repo' }) }
+  let(:checker) { described_class.new(repo) }
+  let(:check_passed) { { result: :passed } }
+  let(:license_text) { '' }
+  let(:license_url) { 'https://raw.githubusercontent.com/dummy-repo/master/LICENSE' }
```
Comment (SteveMarshall) [inline spec/license_checker_spec.rb]: Are we (MOJ) in control of this repo? Or is this a GitHub thing?

File: spec/license_checker_spec.rb [hunk 2]
```diff
@@ -7,43 +7,45 @@
   let(:license_text) { '' }
   let(:license_url) { 'https://raw.githubusercontent.com/dummy-repo/master/LICENSE' }
 
-  it "fetches the license file" do
+  it 'fetches the license file' do
```
Comment (SteveMarshall) [inline spec/license_checker_spec.rb]: These might make more sense to be fixup'd into 62202af?

File: spec/license_checker_spec.rb [hunk 3]
```diff
@@ -7,43 +7,45 @@
...
         expect(checker.run).to eq(check_passed)
       end
     end
 
-    context "when the license has errors" do
+    context 'when the license has errors' do
       let(:license_text) { 'This is not a valid license' }
       let(:check_failed) {
-        {result: :failed, reason: "Not MIT licensed"}
+        {result: :failed, errors: ['Not MIT licensed', 'Not Crown copyright licensed']}
```
Comment (SteveMarshall) [inline spec/license_checker_spec.rb]: 👌

File: lib/repoman/license_checker.rb [hunk 4]
```diff
@@ -2,28 +2,17 @@ module Repoman
...
-      check_content FileFetcher.fetch(license_url)
-    rescue OpenURI::HTTPError => e
-      if e.message =~ /404/
-        add_error "No LICENSE file"
-        result
-      else
-        raise e
-      end
+      has_file, content = RequiredFileChecker.new(repo, 'LICENSE', 'No LICENSE file').fetch
+      has_file ? check_content(content) : add_error(content)
```
Comment (SteveMarshall) [inline lib/repoman/license_checker.rb]: This feels a bit like it's masking the 404 in the name of allowing a neat two line thing here. It also means that the second returned variable from `fetch` is overloaded as both content and error message, which feels… unclean to me?

File: lib/repoman/required_file_checker.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
+module Repoman
+  class RequiredFileChecker
```
Comment (SteveMarshall) [inline lib/repoman/required_file_checker.rb]: Does this need its own tests? Or can some of the `LicenseChecker` tests be made into its tests?

File: spec/license_checker_spec.rb (hunk 1 above)
Comment (digitalronin) [inline spec/license_checker_spec.rb]: There's no such repo, AFAIK. That URL is never used. It's just a value that we expect to be the result of working out what URL we would use to fetch the file called 'LICENSE' from the repo called 'dummy-repo'

File: lib/repoman/required_file_checker.rb (hunk 5 above)
Comment (digitalronin) [inline lib/repoman/required_file_checker.rb]: Good idea. I've added some

File: lib/repoman/license_checker.rb (hunk 4 above)
Comment (digitalronin) [inline lib/repoman/license_checker.rb]: Fair point. What about always returning a 3-element hash from fetch?
```
{ success: true/false, content: [file content]/nil, error: nil/[error message] }
```

**Comment by digitalronin** [comment]
I've separated out the commits that had multiple functions, and reworded the commit messages

File: spec/license_checker_spec.rb (hunk 1 above)
Comment (SteveMarshall) [inline spec/license_checker_spec.rb]: Ah, I see. I wonder if there's a way we can make that more explicit? I thought this was testing against a dummy license file…

File: spec/license_checker_spec.rb (hunk 2 above)
Comment (SteveMarshall) [inline spec/license_checker_spec.rb]: As I mentioned in person…

**Comment by SteveMarshall** [comment]
😎

**Pull request merged**